src/CachePlugin.php

```diff
@@ -67,12 +68,16 @@ public function handleRequest(RequestInterface $request, callable $next, callabl
 
         if ($cacheItem->isHit()) {
             // return cached response
-            return new FulfilledPromise($cacheItem->get());
+            $data = $cacheItem->get();
+            $response = $data['response'];
+            $response = $response->withBody($this->streamFactory->createStream($data['body']));
```

Comment:
does it make sense that the body is always a stream? should we keep what it was when we had a cache miss?